### PR TITLE
Document friendly and legacy rules in ESPHome

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

Document friendly and legacy rules in ESPHome

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/97578
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/esphome.markdown [hunk 1]
```diff
@@ -47,3 +47,46 @@ This integration allows [ESPHome](https://esphome.io) devices to connect directl
 ESPHome devices can make service calls to any [Home Assistant service](https://esphome.io/components/api.html#homeassistant-service-action). This functionality is not enabled by default for newly configured device, but can be turned on the options flow on a per device basis.
 
 {% include integrations/option_flow.md %}
+
+## Entity naming and ids
```
Comment (frenck) [inline source/_integrations/esphome.markdown]:
```suggestion
## Entity naming and IDs
```

File: source/_integrations/esphome.markdown [hunk 2]
```diff
@@ -47,3 +47,46 @@ This integration allows [ESPHome](https://esphome.io) devices to connect directl
...
+
+ESPHome uses different naming and entity id rules based on the configuration of the ESPHome device. It is recommended to set a `friendly_name` in the ESPHome `configuration.yaml` to take advantage of the newer naming structure, which is consistent with Home Assistant naming standards and makes it much easier to tell similar devices apart. The legacy naming rules apply when the `friendly_name` is not set in the `configuration.yaml`.
```
Comment (frenck) [inline source/_integrations/esphome.markdown]:
```suggestion
ESPHome uses different naming and entity ID rules based on the configuration of the ESPHome device. It is recommended to set a `friendly_name` in the ESPHome `configuration.yaml` to take advantage of the newer naming structure, which is consistent with Home Assistant naming standards and makes it much easier to tell similar devices apart. The legacy naming rules apply when the `friendly_name` is not set in the `configuration.yaml`.
```

File: source/_integrations/esphome.markdown [hunk 3]
```diff
@@ -47,3 +47,46 @@ This integration allows [ESPHome](https://esphome.io) devices to connect directl
...
+
+### Friendly naming
+
+- Entity name is a combination of the friendly name and component name
+- Device name is prepended to entity id
+- Entity id uses the ESPHome id
```
Comment (frenck) [inline source/_integrations/esphome.markdown]:
```suggestion
- Entity name is a combination of the friendly name and component name
- Device name is prepended to the entity ID
- Entity ID uses the ESPHome ID
```

File: source/_integrations/esphome.markdown [hunk 4]
```diff
@@ -47,3 +47,46 @@ This integration allows [ESPHome](https://esphome.io) devices to connect directl
...
+
+Example:
+
+```yaml
+esphome:
+   name: livingroomdesk
+   friendly_name: Living Room Desk
```
Comment (frenck) [inline source/_integrations/esphome.markdown]:
```suggestion
   name: "livingroomdesk"
   friendly_name: "Living room desk"
```

File: source/_integrations/esphome.markdown [hunk 5]
```diff
@@ -47,3 +47,46 @@ This integration allows [ESPHome](https://esphome.io) devices to connect directl
...
+
+sensor:
+   name: Temperature
+   id: temperature
```
Comment (frenck) [inline source/_integrations/esphome.markdown]:
```suggestion
   name: "Temperature"
   id: "temperature"
```

File: source/_integrations/esphome.markdown [hunk 6]
```diff
@@ -47,3 +47,46 @@ This integration allows [ESPHome](https://esphome.io) devices to connect directl
...
+```
+
+The entity will be named `Living Room Desk Temperature` and will default to having an entity_id of `sensor.livingroomdesk_temperature`
```
Comment (frenck) [inline source/_integrations/esphome.markdown]:
```suggestion
The entity will be named `Living room desk Temperature` and will default to having an entity ID of `sensor.livingroomdesk_temperature`.
```

File: source/_integrations/esphome.markdown [hunk 7]
```diff
@@ -47,3 +47,46 @@ This integration allows [ESPHome](https://esphome.io) devices to connect directl
...
+
+### Legacy naming
+
+- Entity name is the component name
+- Device name is not prepended to the entity name
+- Entity id is derived solely from the entity name
```
Comment (frenck) [inline source/_integrations/esphome.markdown]:
```suggestion
- Entity ID is derived solely from the entity name
```

File: source/_integrations/esphome.markdown [hunk 8]
```diff
@@ -47,3 +47,46 @@ This integration allows [ESPHome](https://esphome.io) devices to connect directl
...
+
+Example:
+
+```yaml
+esphome:
+   name: livingroomdesk
```
Comment (frenck) [inline source/_integrations/esphome.markdown]:
```suggestion
   name: "livingroomdesk"
```

File: source/_integrations/esphome.markdown [hunk 9]
```diff
@@ -47,3 +47,46 @@ This integration allows [ESPHome](https://esphome.io) devices to connect directl
...
+
+sensor:
+   name: Temperature
+   id: temperature
```
Comment (frenck) [inline source/_integrations/esphome.markdown]:
```suggestion
   name: "Temperature"
   id: "temperature"
```

File: source/_integrations/esphome.markdown [hunk 10]
```diff
@@ -47,3 +47,46 @@ This integration allows [ESPHome](https://esphome.io) devices to connect directl
...
+```
+
+The entity will be named `Temperature` and will default to having an entity_id of `sensor.temperature`
```
Comment (frenck) [inline source/_integrations/esphome.markdown]:
```suggestion
The entity will be named `Temperature` and will default to having an entity_id of `sensor.temperature`.
```

**Review by frenck — APPROVED** [review]
Thanks, @bdraco 👍 

../Frenck

**Comment by bdraco** [comment]
Thanks and thanks for the tweaks ❤️

**Pull request merged**